fendermint/vm/topdown/src/finality/null.rs

```diff
@@ -157,14 +158,16 @@ impl FinalityWithNull {
 /// All the private functions
 impl FinalityWithNull {
     /// Get the first non-null block in the range [start, end].
-    fn min_nonnull_block(&self, start: BlockHeight, end: BlockHeight) -> Stm<Option<BlockHeight>> {
+    fn first_non_null_block_before(&self, height: BlockHeight) -> Stm<Option<BlockHeight>> {
         let cache = self.cached_data.read()?;
-        for h in start..=end {
-            if let Some(Some(_)) = cache.get_value(h) {
-                return Ok(Some(h));
+        Ok(cache.lower_bound().and_then(|lower_bound| {
```

Comment:
I assume here that if the cache doesn't have the height it will go to fetch it from the parent, right?